### PR TITLE
Fix tests buck build when running locally

### PR DESCRIPTION
**Pull request by janicduplessis**
Got this error when trying to run instrumentation tests locally with `./scripts/run-android-local-integration-test.sh`

```
C:\Users\janic\Developer\react-native\ReactAndroid\src\androidTest\java\com\facebook\react\tests\DatePickerDialogTestCase.java:110: error: cannot access com.facebook.react.modules.core.DefaultHardwareBackBtnHandler
    return (DialogFragment) getActivity().getSupportFragmentManager()
                                         ^
  class file for com.facebook.react.modules.core.DefaultHardwareBackBtnHandler not found
C:\Users\janic\Developer\react-native\ReactAndroid\src\androidTest\java\com\facebook\react\tests\ViewRenderingTestCase.java:82: error: cannot access com.facebook.react.touch.ReactInterceptingViewGroup
    assertEquals("Incorrect (or not applied) opacity", expectedOpacity, view.getAlpha());
                                                                            ^
  class file for com.facebook.react.touch.ReactInterceptingViewGroup not found
C:\Users\janic\Developer\react-native\ReactAndroid\src\androidTest\java\com\facebook\react\tests\ViewRenderingTestCase.java:86: error: cannot access com.facebook.react.touch.ReactHitSlopView
        view.getBackgroundColor());
            ^
  class file for com.facebook.react.touch.ReactHitSlopView not found
C:\Users\janic\Developer\react-native\ReactAndroid\src\androidTest\java\com\facebook\react\tests\ReactHorizontalScrollViewTestCase.java:-1: note: Some input files use or override a deprecated API.

C:\Users\janic\Developer\react-native\ReactAndroid\src\androidTest\java\com\facebook\react\tests\ReactHorizontalScrollViewTestCase.java:-1: note: Recompile with -Xlint:deprecation for details.

Errors: 3. Warnings: 0.

BUILD FAILED: //ReactAndroid/src/androidTest/java/com/facebook/react/tests:tests failed with exit code 1:
javac
[-] BUILDING...FINISHED 52.2s [100%]
Not using buckd because NO_BUCKD is set.
```

Not sure exactly why it works on CI but adding those 2 lines to the tests BUCK file fixes the issue. Also added the generated test bundle to gitignore.

**Test plan**
Ran integration tests locally with `./scripts/run-android-local-integration-test.sh`

### CODE REVIEW

**Comment by janicduplessis** [comment]
cc @bestander

**Comment by facebook-github-bot-4** [comment]
By analyzing the blame information on this pull request, we identified @bestander and @mkonicek to be potential reviewers.

File: ReactAndroid/src/androidTest/java/com/facebook/react/tests/BUCK [hunk 1]
```diff
@@ -14,6 +14,8 @@ deps = [
   react_native_target('java/com/facebook/react/bridge:bridge'),
   react_native_target('java/com/facebook/react:react'),
   react_native_target('java/com/facebook/react/common:common'),
+  react_native_target('java/com/facebook/react/modules/core:core'),
+  react_native_target('java/com/facebook/react/touch:touch'),
```
Comment (bestander) [inline ReactAndroid/src/androidTest/java/com/facebook/react/tests/BUCK]: could  you move this down a bit to preserve alphabetic ordering please

**Comment by bestander** [comment]
Seems legit, let's ship it, just one little nit

**Comment by mkonicek** [comment]
👍  Thanks @janicduplessis.

**Comment by facebook-github-bot-7** [comment]
@janicduplessis updated the pull request.

**Comment by bestander** [comment]
@facebook-github-bot shipit

**Comment by facebook-github-bot-6** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://our.intern.facebook.com/intern/opensource/github/pull_request/459109967631739/int_phab) to review.

**Pull request closed without merging**